### PR TITLE
Home taks 4

### PR DESCRIPTION
**Pull request by Shtuka1993**

### CODE REVIEW

File: Crud.php [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+            $numberEntity->setSlug($slug);
+            $numberEntity->setTitle($title);
+            $numberEntity->setNumber($number);
+            $numberEntity->setText($text);
+            $numberEntity->setTranscription($transcription);
+            $numberEntity->setDate(new DateTime());
+            $this->entityManager->persist($numberEntity);
+            $this->entityManager->flush();
+
+            return true;
```
Comment (vpashkov-cartera) [inline Crud.php:46]: what is the purpose of return true;?

File: Crud.php [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        /**
+         * Search numbers
+         * 
+         * @param slug
+         * @param title
+         * @param text
+         * @param transcription
+         * @param date
+         */
+        public function findNumbers($slug = null, $title = null, $text = null, $transcription = null, $date = null) {}
```
Comment (vpashkov-cartera) [inline Crud.php]: ??? empty method?

File: Crud.php [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+        public function readNumbersPaginate(int $page = 1):array {               
+            // get the user repository
+            $numbers =  $this->entityManager->getRepository(Number::class);
+
+            // build the query for the doctrine paginator
+            $query = $numbers->createQueryBuilder('n')
+                ->getQuery();
+
+            //set page size
+            $pageSize = self::PER_PAGE;
```
Comment (vpashkov-cartera) [inline Crud.php]: I believe, $pageSize variable here is redundund.  You could use self:PER_PAGE constant here.

File: Crud.php [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+         */
+        public function findNumbers($slug = null, $title = null, $text = null, $transcription = null, $date = null) {}
+
+        /**
+         * Read all numbers
+         * 
+         * @return object
+         */
+        public function readAllNumbers():object {
+            $numberRepository = $this->entityManager->getRepository('Number');
```
Comment (vpashkov-cartera) [inline Crud.php]: What is the difference between `$this->entityManager->getRepository('Number')` and `$this->entityManager->getRepository(Number::class)`

File: Crud.php [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+            $numbers =  $this->entityManager->getRepository(Number::class);
+
+            // build the query for the doctrine paginator
+            $query = $numbers->createQueryBuilder('n');
+
+            //setting sorting
+            $query = $query->orderBy('n.slug', $sorting);
+
+            //Adding searching parametters
+            if(!empty($filters['slug'])) {
```
Comment (vpashkov-cartera) [inline Crud.php:148]: Smells like code duplication line 137 and line 190. Please try to refactor it

File: Crud.php [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+         * @param string slug 
+         * @param string title
+         * @param int number 
+         * @param string text 
+         * @param string transcription
+         * 
+         * @return bool 
+         */
+        public function updateNumber(int $id, string $slug, string $title, int $number, string $text, string $transcription):bool {
+            $numberEntity = $this->entityManager->find('Number', $id);
```
Comment (vpashkov-cartera) [inline Crud.php]: is it possible to extract updateNumber and createNumber  common functionality?

File: Crud.php (hunk 1 above)
Comment (Shtuka1993) [reply Crud.php:46]: I was thinking that it would be better if we have returning data from function.

File: Crud.php (hunk 4 above)
Comment (Shtuka1993) [reply Crud.php]: No difference. Will use Number::class because it use internal data from class not some magic string

File: Crud.php (hunk 2 above)
Comment (Shtuka1993) [reply Crud.php]: Will remove. Forgot to delete.

File: Crud.php (hunk 3 above)
Comment (Shtuka1993) [reply Crud.php]: Yes I created that constant for this purpose. Will change.

File: Crud.php (hunk 5 above)
Comment (Shtuka1993) [reply Crud.php:148]: Will refactor read number paginated

File: Crud.php (hunk 6 above)
Comment (Shtuka1993) [reply Crud.php]: Extracted to Entity

**Review by vpashkov-cartera — APPROVED** [review]

**Pull request merged**